src/java.base/share/classes/java/lang/StackWalker.java

```diff
@@ -39,6 +39,21 @@
 /**
  * A stack walker.
  *
+ * <p> The {@link Kind} specifies the information of the stack frames to be collected.
+ * The {@link #getInstance() getInstance} factory methods return a {@code StackWalker}
+ * that collects the {@linkplain Kind#METHOD_INFO method information}
+ * from the stack frames by default.  If the method information is not needed,
+ * a {@code StackWalker} collecting {@linkplain Kind#CLASS_INFO class only information}
+ * can be used instead via the {@link #getInstance(Kind, Option...)} method.
```

Comment:
The description of `Kind` should come after the basics of what StackWalker does:
```
 * <p> The {@link StackWalker#walk walk} method opens a sequential stream
 * of {@link StackFrame StackFrame}s for the current thread and then applies
 * the given function to walk the {@code StackFrame} stream.
```

It should be clarified that `METHOD_INFO` returns method information _as well as_ class information.

Overall I would still prefer adding an `Option.OMIT_METHOD_INFO` over a new `Kind` enum.